### DISCUSSION TITLE
Is there any workarounds to refer string labels?

### DISCUSSION THREAD

**Discussion by FogDong** [cue-lang/cue#1770] [Q&A] [answered]
If `a` has a property `a-b`, then we can use `a."a-b"` like below:

```cue
a: {
  "a-b": "aaa"
}

result1: a["a-b"]
```

However, we can not refer a string label like:
```
"c-d": {
  b: "bbb"
}

result2: "c-d".b
```

I know that we can use alias like:
```
alias="a-b":{
  b: "bbb"
}

e: alias.b
```

However, in my situation, the label name might be a uuid which means that it can start with number, and users like to use the uuid to refer to values, so I'm wondering is there any workarounds to refer string labels?

Thanks in advance.

**Comment by rogpeppe** [comment, marked as answer]
One possibility is to move all those top level entities into a hidden field, and then embed that.

For example:


```
_top: {
    "c-d": {
        b: "bbb"
    }
}
_top

result2: _top."c-d".b
```

**Reply by FogDong** [reply]
I think this solution is quite similar to the `alias`, my final solution to this question is to add a readable field for users to access.
Thanks for the reply :)

**Reply by rogpeppe** [reply]
The main difference as I see it is that with this solution, you can define just one alias rather than an alias for each field which could be tedious and error-prone, particularly if you're naming the alias after the UUID that it's attached to.
